config/dex.yaml

```diff
@@ -0,0 +1,55 @@
+issuer: '{{ getenv "APP_HOME_URL" }}/dex'
+
+storage:
+  type: sqlite3
+  config:
+    file: /var/dex/dex.db
+
+web:
+  http: '0.0.0.0:5556'
+
+logger:
+  level: "debug"
+  format: "text"
+
+frontend:
+  issuer: "Grist"
+  logoURL: '{{ getenv "APP_HOME_URL" }}/v/unknown/ui-icons/Logo/GristLogo.svg'
+
+staticClients:
+- id: '{{ getenv "GRIST_OIDC_IDP_CLIENT_ID" }}'
+  redirectURIs:
+  - '{{ getenv "APP_HOME_URL" }}/oauth2/callback'
+  name: 'Grist'
+  secret: '{{ getenv "GRIST_OIDC_IDP_CLIENT_SECRET" }}'
+
+
+oauth2:
+  skipApprovalScreen: true
+
+connectors:
+  - type: oidc
+    id: authelia
+    name: Basic Username
+    config:
+      issuer: '{{ getenv "APP_HOME_URL" }}/authelia'
+      clientID: 'dex-local'
+      clientSecret: '{{ getenv "DEX_CLIENT_SECRET" }}'
+      redirectURI: '{{ getenv "APP_HOME_URL" }}/dex/callback'
+
+  - type: google
+    id: google
+    name: Google
+    config:
+      issuer: https://accounts.google.com
+      clientID: '{{ getenv "GOOGLE_CLIENT_ID" }}'
+      clientSecret: '{{ getenv "GOOGLE_CLIENT_SECRET" }}'
+      redirectURI: '{{ getenv "APP_HOME_URL" }}/dex/callback'
+
+  - type: microsoft
+    id: microsoft
+    name: Microsoft
+    config:
+      clientID: '{{ getenv "MICROSOFT_CLIENT_ID" }}'
+      clientSecret: '{{ getenv "MICROSOFT_CLIENT_SECRET" }}'
+      redirectURI: '{{ getenv "APP_HOME_URL" }}/dex/callback'
```

Comment:
We should add these environment variables to env-template.

Better yet, could we make these options only appear if the environment variables set?

Go templates (used by dex) seem to support `if` `end` syntax for conditionally including template content:
https://developer.hashicorp.com/nomad/tutorials/templates/go-template-syntax#control-structure-list

Reply:
Good idea, I made these conditional on the `ID` variables being nonempty.